packages/opentelemetry-tracing/src/BasicTracerProvider.ts

```diff
@@ -42,13 +41,11 @@ export class BasicTracerProvider implements api.TracerProvider {
   readonly logger: api.Logger;
   readonly resource: Resource;
 
-  constructor(config: TracerConfig = DEFAULT_CONFIG) {
-    this.logger = config.logger ?? new ConsoleLogger(config.logLevel);
-    this.resource = config.resource ?? Resource.createTelemetrySDKResource();
-    this._config = Object.assign({}, config, {
-      logger: this.logger,
-      resource: this.resource,
-    });
+  constructor(config?: TracerConfig) {
+    const mergedConfig = { ...DEFAULT_CONFIG, ...config };
```

Comment:
Quick check that this is intentionally a shallow merge correct?

Reply:
Good call. It was not intentional per se, but it works as confit options are seemingly 1 level deep ...should we use lodash's merge instead?

Reply:
I would avoid adding lodash to just have deep merge from that library, we can have just our own function that does that

Reply:
FWIW loadash allows us to only import the function we need, not lodash entirely

Reply:
The point is to keep our set of dependencies as small as possible, which in turn decreases the dependency burden on our customers. I would prefer to either keep the shallow merge or write our own deep merge.

Reply:
Ok makes sense. Shallow merging suffices today so I'll keep it as is if no objections.

Reply:
Since today it does a shallow merge anyways this isnt a loss in functionality i'm fine with it. if someone wants to tackle deep merge or we get a customer that asks for it we can always add later.

Reply:
Have created an issue for that as it is coming back and haven't been done yet. https://github.com/open-telemetry/opentelemetry-js/issues/1496

Reply:
@naseemkullah After thinking a while, I think having a proper function for doing a deep merge of 2 configs would be really essential, especially after this change. Would that be possible in this PR as well ?, could you try to implement or port such functionality ?

Reply:
@obecny I've decided to create a separate PR for the util: https://github.com/open-telemetry/opentelemetry-js/pull/1503/files is that ok?